lib/devices/Roborock.js

```diff
@@ -335,18 +344,67 @@ class Roborock extends MiioVacuum {
         await this.sendLocal("set_timezone", [new_zone], {});
     }
 
-    /*
-        0-100: percent
-
-        Or presets:
-        101: quiet
-        102: balanced
-        103: Turbo
-        104: Max
-        105: Mop
+    /**
+     * @private
+     */
+    detectFanSpeeds(msg_ver) {
+        let fanSpeeds = {
+            "whisper": {label: "Whisper", value: 1},
+            "min": {label: "Quiet", value: 38},
+            "medium": {label: "Balanced", value: 60},
+            "high": {label: "Turbo", value: 75},
+            "max": {label: "Max", value: 100},
+            "mop": {label: "Mop", value: 105},
```

Comment:
Are you sure you want to model mopping as a "fan-speed"?

Reply:
This sadly is how roborock did it :/

Reply:
I understand, but that’s something I think the Valetudo API could abstract away right?
Roborock.js can know about this, but in the UI the mopping option should probably not be available via the fan-speed button?
* For Viomi I added support to enable mopping automatically based on installed box (#474). Not sure if something like that would be feasible for RR as well?
* For Viomi there are two settings, mixed (mop&vacuum) or mop only, which is currently not exposed through the UI, so a mopping-settings UI would still make sense.

(We can file this a feature request and leave this out of this refactoring for now).

Reply:
I think you should propose the webinterface/API changes for viomi and then we look to adapt the roborock implementation.